integrations/telegram/src/misc/utils.ts

```diff
@@ -0,0 +1,64 @@
+import { axios } from '@botpress/client'
+import { PICTURE_LIMIT_SIZE } from 'src/const'
+import { Telegraf } from 'telegraf'
+import { PhotoSize, User } from 'telegraf/typings/core/types/typegram'
+
+export const getUserNameFromTelegramUser = (telegramUser: User) => {
+  if (telegramUser.last_name) {
+    return `${telegramUser.first_name} ${telegramUser.last_name}`
+  } else if (telegramUser.username) {
+    return `${telegramUser.first_name} (username: ${telegramUser.username})`
+  }
+  return `${telegramUser.first_name} (id: ${telegramUser.id})`
+}
+
+const getDataUriFromImgHref = async (imgHref: string): Promise<string> => {
+  const fileExtension = imgHref.substring(imgHref.lastIndexOf('.') + 1)
+
+  const { data } = await axios.default.get(imgHref, { responseType: 'arraybuffer' })
+
+  const base64File = Buffer.from(data, 'binary').toString('base64')
+
+  return `data:image/${fileExtension};base64,${base64File}`
+}
+
+const getBestPhotoSize = (photos: PhotoSize[]): PhotoSize | null =>
+  photos.reduce((finalPicture: PhotoSize | null, currentPicture: PhotoSize) => {
```

Comment:
Cool approach 🤓 Although I have a hunch this code could be simpler and easier to read in the future with just a `for` loop and a `break` when the first image below the dimensions/filesize is found, otherwise return `null` at the end of the loop, feasible? 😄

Reply:
yes I can do that but the point of using a reduce here is taking the highest resolution available below the limit, still need to loop over them all, I can still do that with a for loop though